test/functional/lib/tests.cpp

```diff
@@ -2194,22 +2200,25 @@ CreateTcpSocket(
     // within the timeout, the listening socket gets closed, which will cancel
     // the accept request.
     //
-    auto AsyncThread = std::async(
-        std::launch::async,
-        [&] {
-            SOCKADDR_INET Address;
-            INT AddressLength = sizeof(Address);
-            return unique_fnsock(FnSockAccept(Socket.get(), (SOCKADDR *)&Address, &AddressLength));
-        }
-    );
+    TCP_ACCEPT_THREAD_CONTEXT Ctx;
+    Ctx.ListeningSocket = Socket.get();
+    CXPLAT_THREAD_CONFIG ThreadConfig {
+        0, 0, NULL, TcpAcceptFn, &Ctx
+    };
+    CXPLAT_THREAD AsyncThread;
+    TEST_TRUE(CXPLAT_SUCCEEDED(CxPlatThreadCreate(&ThreadConfig, &AsyncThread)));
+    auto DeleteThread = wil::scope_exit([&]
+    {
+        CxPlatThreadDelete(&AsyncThread);
+    });
```

Comment:
Not blocking, but opportunity to replace `CXPLAT_THREAD` with a C++ wrapper. This is what I have in MsQuic:
```c++
class CxPlatThread {
    CXPLAT_THREAD Thread {0};
    bool Initialized : 1;
    bool WaitOnDelete : 1;
public:
    CxPlatThread(bool WaitOnDelete = true) noexcept : Initialized(false), WaitOnDelete(WaitOnDelete) { }
    ~CxPlatThread() noexcept {
        if (Initialized) {
            if (WaitOnDelete) {
                CxPlatThreadWait(&Thread);
            }
            CxPlatThreadDelete(&Thread);
        }
    }
    QUIC_STATUS Create(CXPLAT_THREAD_CONFIG* Config) noexcept {
        auto Status = CxPlatThreadCreate(Config, &Thread);
        if (QUIC_SUCCEEDED(Status)) {
            Initialized = true;
        }
        return Status;
    }
    void Wait() noexcept {
        if (Initialized) {
            CxPlatThreadWait(&Thread);
        }
    }
};
```